chef/cookbooks/provisioner/templates/suse/crowbar_register.erb

```diff
@@ -72,6 +72,51 @@ if test $CROWBAR_FORCE -ne 1; then
     fi
 fi
 
+# Helper functions
+# ----------------
+
+post_allocate() {
+  local curlargs=(--connect-timeout 60 -s \
+                  -L -X POST --data-binary "{ \"name\": \"$1\" }" \
+                  -H "Accept: application/json" -H "Content-Type: application/json" \
+                  --max-time 240)
+
+  [[ $CROWBAR_KEY ]] && curlargs+=(-u "$CROWBAR_KEY" --digest --anyauth)
+
+  curl "${curlargs[@]}" "http://$ADMIN_IP:3000/crowbar/machines/1.0/allocate/0"
+}
+
+post_state() {
+  local curlargs=(-o "/var/log/crowbar/$1-$2.json" --connect-timeout 60 -s \
+                  -L -X POST --data-binary "{ \"name\": \"$1\", \"state\": \"$2\" }" \
+                  -H "Accept: application/json" -H "Content-Type: application/json" \
+                  --max-time 240)
+
+  [[ $CROWBAR_KEY ]] && curlargs+=(-u "$CROWBAR_KEY" --digest --anyauth)
+
+  curl "${curlargs[@]}" "http://$ADMIN_IP:3000/crowbar/crowbar/1.0/transition/default"
+}
+
+add_group() {
+  gid=$(getent group $1 | cut -f 3 -d ":")
+  if [[ -z "$gid" ]]; then
+    groupadd --system --gid $2 $1
+  elif [[ "$gid" != "$2" ]]; then
+    echo "Group $1 did not have gid $2."
+    exit 1
```

Comment:
I would say exit, changing it automatically can break other parts of the system due to we do not change the ID for existing files.